File: Data Structures & Algorithms/is-anagram/submission-0.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if (len(s) != len(t)):
            return False

        letterT = {}
        letterS = {}

        for i in range(len(s)):
            if s[i] in letterS:
                letterS[s[i]] = 1 + letterS[s[i]]
            else:
                letterS[s[i]] = 1
            
            if t[i] in letterT:
                letterT[t[i]] = 1 + letterT[t[i]]
            else:
                letterT[t[i]] = 1
        
        return letterT == letterS

        

File: Data Structures & Algorithms/palindromic-substrings/submission-0.cpp
class Solution {
public:
    // at each letter: either pick it or not
    // if we pick a letter, 
    // let dp[i] be the max number of palindromes from letters 1..i
    // dp

    int countSubstrings(string s) {
        int res = 0;

        for (int i = 0; i < s.size(); i++) {
            int l = i;
            int r = i;

            while (l >= 0 && r < s.size() && s[l] == s[r]) {
                res++;
                l--;
                r++;
            }

            l = i;
            r = i+1;
            while (l >= 0 && r < s.size() && s[l] == s[r]) {
                res++;
                l--;
                r++;
            }
        }

        return res;
    }
};
